### PR TITLE
ToInt64 now throws exception when appropriate

### PR DESCRIPTION
**Pull request by MateuszKlatecki**
## Description
Convert function didn't throw exceptions if value in input string isn't correct number or is out of range and returns 0 in above situation. After this change exceptions are thrown.

## Motivation and Context

- Fixes problem with incorrect parse integers

## How Has This Been Tested?

I wrote a simple program that tests various possibilities. I can add it to example programs.
There is still problem during UInt64 and Int64 Parse. If value is out of range function didn't throw exception but return maximal or minimal allowed value.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

Signed-off-by: GITHUB_USER <GITHUB_USER_EMAIL>

### CODE REVIEW

File: src/CLR/CorLib/corlib_native_System_Convert.cpp [hunk 1]
```diff
@@ -19,36 +19,72 @@ HRESULT Library_corlib_native_System_Convert::NativeToInt64___STATIC__I8__STRING
 
       #if (SUPPORT_ANY_BASE_CONVERSION == TRUE)
         // suport for conversion from any base
+        char* endptr = NULL;
 
-        int64_t zero = 0;
+        bool UInt64 = false;
```
Comment (josesimoes) [inline src/CLR/CorLib/corlib_native_System_Convert.cpp]: This var name can be confused with a type. Better to rename to... `isUInt64` for example.

File: src/CLR/CorLib/corlib_native_System_Convert.cpp [hunk 2]
```diff
@@ -19,36 +19,72 @@ HRESULT Library_corlib_native_System_Convert::NativeToInt64___STATIC__I8__STRING
...
+
+        if (UInt64) {
+            // for UInt64 the check will be bypassed
+            stack.SetResult_I8(result);
+        } else if (!isSigned && maxValue != 0x7FFFFFFFFFFFFFFF && (uint64_t)result > (uint64_t)maxValue) {
+            NANOCLR_SET_AND_LEAVE(CLR_E_OUT_OF_RANGE);
+        } else if (result > maxValue || result < minValue) {
+            NANOCLR_SET_AND_LEAVE(CLR_E_OUT_OF_RANGE);
+        } else {
+            stack.SetResult_I8(result);
```
Comment (josesimoes) [inline src/CLR/CorLib/corlib_native_System_Convert.cpp]: The `else` here is setting the return value like the `if (UInt64)`  above.
Is this the intended way? Can't they be grouped? 
A comment explaining the reasoning behind it is in order.

**Comment by josesimoes** [comment]
@MateuszKlatecki yes please add some test code to the samples.
Also copy/paste the output in the PR comment above, please. Just to keep a record of the output.

File: src/CLR/CorLib/corlib_native_System_Convert.cpp (hunk 1 above)
Comment (MateuszKlatecki) [reply src/CLR/CorLib/corlib_native_System_Convert.cpp]: Changed

File: src/CLR/CorLib/corlib_native_System_Convert.cpp (hunk 2 above)
Comment (MateuszKlatecki) [reply src/CLR/CorLib/corlib_native_System_Convert.cpp]: You are right, this can be simplified.

**Pull request merged**